### PR TITLE
Keyboard Tabbing - HTML Mobile Masthead

### LINKED ISSUE #1578: Enhance Keyboard Tabbing - HTML Mobile Masthead
**User Story:**
As a Spark we want small viewport menu links to be highlighted when selected with keyboard tab so that it's accessibility friendly. 

**Notes:**
- Requestor: RV while developing Manual Testing document
- Angular & React are working as expected.

**AC:**
- small viewport menu links will be highlighted when selected with keyboard tab 
(*will be the same as hover interaction)

image

### PR DESCRIPTION
**Pull request by guern1kn**
## What does this PR do?
Updates the keyboard event for the HTML Mobile Masthead. 
- Update the selector for the mobile masthead to be the menu button.
- Remove `data-sprk-main` from tests and stories for masthead, it's not longer required for masthead.
- Update the isOpen variable to update on click of the menu button.
- Switch the focusin event to be a click event.
- Wrap the focusTrap function in a timeout to add it to the call stack after the mobile menu open/close toggle javascript is finished.
- Update masthead tests.

### Associated Issue
Fixes #1578 

### Code
 - [x] Build Component in HTML
 - [x] Unit Testing in HTML with `npm run test` in `html/` (100% coverage, 100% passing)

### Accessibility
- [x] New changes abide by [accessibility requirements](https://sparkdesignsystem.com/docs/accessibility)

### Browser Testing (current version and 1 prior)
  - [x] Google Chrome
  - [x] Mozilla Firefox
  - [x] Microsoft Internet Explorer 11 (only this specific version of IE)
  - [x] Microsoft Edge
  - [x] Apple Safari

### CODE REVIEW

File: html/tests/masthead.tests.js [hunk 1]
```diff
@@ -308,14 +309,6 @@ describe('masthead UI Events tests', () => {
...
   });
 
-  it('should focus on the first nav item when focusin is triggered on a narrow viewport when the nav is open', () => {
-    iconContainer.focus();
-    nav.classList.add('sprk-c-Masthead__narrow-nav');
-    event = new window.Event('focusin');
-    main.dispatchEvent(event);
-    expect(document.activeElement).eql(navItem);
-  });
-
```
Comment (guern1kn) [inline html/tests/masthead.tests.js]: Coverage is maintained while deleting this test.

File: html/components/masthead.js [hunk 2]
```diff
@@ -116,7 +116,7 @@ const showSelectorMask = mastheadSelectorMask => {
 
 const bindUIEvents = () => {
   getElements('[data-sprk-mobile-nav-trigger]', element => {
-    const mainLayout = document.querySelector('[data-sprk-main]');
```
Comment (afebbraro) [inline html/components/masthead.js]: with this update, what happens to the [data-sprk-main] selector? does the masthead no longer need it?

File: html/components/masthead.js [hunk 3]
```diff
@@ -116,7 +116,7 @@ const showSelectorMask = mastheadSelectorMask => {
...
+    const mobileMenu = document.querySelector('.sprk-c-Menu');
```
Comment (afebbraro) [inline html/components/masthead.js]: We prefer selecting on data attributes vs classes as much as possible.

File: html/components/masthead.js (hunk 2 above)
Comment (guern1kn) [reply html/components/masthead.js]: Yeah, good catch. We don't use that attribute for anything else in the masthead. I've updated the story, tests, PR to reflect it.

**Pull request merged**